File: Project_Sourcecode_go/client/Account1.go
package client

import (
	"cs586_project/abstractfactory"
	"cs586_project/datastore"
	"cs586_project/state"
)

type Account1 struct {
	m  *state.MdaEfsm
	ds interface{}
}

func (a1 *Account1) Init(m *state.MdaEfsm, factory abstractfactory.AbstractFactoryApi) {
	a1.m = m
	a1.ds = factory.GetDs()
}

func (a1 *Account1) Open(p int, y int, a int) {
	ds1 := a1.ds.(*datastore.DS1)
	ds1.Temp_p = p
	ds1.Temp_y = y
	ds1.Temp_a = a

	a1.m.Open()
}

func (a1 *Account1) Pin(x int) {
	ds := a1.ds.(*datastore.DS1)
	if x == ds.Pin {
		if ds.Balance > 100 {
			a1.m.CorrectPinAboveMin()
		} else {
			a1.m.CorrectPinBelowMin()
		}
	} else {
		a1.m.IncorrectPin(1)
	}
}

func (a1 *Account1) Deposit(d int) {
	ds := a1.ds.(*datastore.DS1)
	ds.Temp_d = d
	a1.m.Deposit()
	if ds.Balance > 100 {
		a1.m.AboveMinBalance()
	} else {
		a1.m.BelowMinBalance()
	}
}

func (a1 *Account1) Withdraw(w int) {
	ds := a1.ds.(*datastore.DS1)
	ds.Temp_w = w
	a1.m.Withdraw()
	if ds.Balance > 100 {
		a1.m.AboveMinBalance()
	} else {
		a1.m.WithdrawBelowMinBalance()
	}
}

func (a1 *Account1) Balance() {
	a1.m.Balance()
}

func (a1 *Account1) Login(y int) {
	ds := a1.ds.(*datastore.DS1)
	if y == ds.Id {
		a1.m.Login()
	} else {
		a1.m.IncorrectLogin()
	}
}

func (a1 *Account1) Logout() {
	a1.m.Logout()
}

func (a1 *Account1) Lock(x int) {
	ds := a1.ds.(*datastore.DS1)
	if x == ds.Pin {
		a1.m.Lock()
	} else {
		a1.m.IncorrectLock()
	}
}

func (a1 *Account1) Unlock(x int) {
	ds := a1.ds.(*datastore.DS1)
	if x == ds.Pin {
		a1.m.Unlock()
		if ds.Balance > 100 {
			a1.m.AboveMinBalance()
		} else {
			a1.m.BelowMinBalance()
		}
	} else {
		a1.m.IncorrectUnlock()
	}
}

File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/input/Account1.java
package edu.iit.cs586.project.input;

import edu.iit.cs586.project.MDA_EFSM;
import edu.iit.cs586.project.abstractfactory.AbstractFactory;
import edu.iit.cs586.project.datastore.DS1;
import edu.iit.cs586.project.datastore.DataStore;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/1 20:28
 */
public class Account1 {
    private MDA_EFSM m;
    private DataStore ds;

    public Account1(AbstractFactory af, MDA_EFSM m) {
        this.ds = af.getDataStore();
        this.m = m;
    }

    public void open(int p, int y, int a) {
        ds.temp_p = p;
        ds.temp_y = y;
        ((DS1) ds).temp_a = a;

        m.Open();
    }

    public void pin(int x) {
        if (x == ds.pn) {
            if (((DS1) ds).b > 100) {
                m.CorrectPinAboveMin();
            } else {
                m.CorrectPinBelowMin();
            }
        } else {
            m.IncorrectPin(1);
        }
    }

    public void deposit(int d) {
        ((DS1) ds).temp_d = d;
        m.Deposit();
        if (((DS1) ds).b > 100) {
            m.AboveMinBalance();
        } else {
            m.BelowMinBalance();
        }
    }

    public void withdraw(int w) {
        ((DS1) ds).temp_w = w;
        m.Withdraw();
        if (((DS1) ds).b > 100) {
            m.AboveMinBalance();
        } else {
            m.WithdrawBelowMinBalance();
        }
    }

    public void balance() {
        m.Balance();
    }

    public void login(int y) {
        if (y == ds.id) {
            m.Login();
        } else {
            m.IncorrectLogin();
        }
    }

    public void logout() {
        m.Logout();
    }

    public void lock(int x) {
        if (x == ds.pn) {
            m.Lock();
        } else {
            m.IncorrectLock();
        }

    }

    public void unlock(int x) {
        if (x == ds.pn) {
            m.Unlock();
            if (((DS1) ds).b > 100) {
                m.AboveMinBalance();
            } else {
                m.BelowMinBalance();
            }
        } else {
            m.IncorrectUnlock();
        }
    }
}
